src/message.rs

```diff
@@ -302,6 +302,16 @@ impl MessageKind {
         }
     }
 
+    pub fn get_amount(&self) -> Option<Amount> {
+        if self.action != Action::TakeSell && self.action != Action::TakeBuy {
```

Comment:
Only buyers can use the variant PaymentRequest, so this condition should be `self.action != Action::TakeSell` only

Reply:
Right!